runtime-modules/forum/src/lib.rs

```diff
@@ -1414,24 +1440,72 @@ impl<T: Trait> Module<T> {
         }
     }
 
+    fn ensure_can_delete_category(
```

Comment:
This appears to not allow the lead to act as a normal moderator, is that right?

There is something wrong here beyond that, because you seem to be checking that the moderator possibly is the lead, which is not possible. The lead does not have a moderator id.

Reply:
No only moderator can delete category right now. I only wanted to provide the possibility for lead to deleting the root category. You are right, moderator + lead checks together are invalid, I didn't realize that before.

Should I create separate function `delete_category_lead` to be used by lead only (on any category?), or should I let update the function to be able to accept both lead and moderator (via union type `delete_category(..., moderator_or_lead: UnionTypeOfModeratorAndLead)`)?

Reply:
Your last suggestion is the right approach, just make the names more sensible, something like

```Rust
enum PrivilegedActor{
Lead,
Moderator(T::ModeratorId)
}
```

Now you can reuse this type for multiple extrinsic where you may need to allow both the lead and the moderator to act.

Well done on identifying the right approach right away ;)